src/app/shared/text-overflow/text-overflow-ellipses.component.ts

```diff
@@ -0,0 +1,17 @@
+import { Component, Input } from '@angular/core';
+
+@Component({
+  selector: 'publish-text-overflow-ellipses',
+  template: `
+    <div [style.max-height]="(numLines * lineHeight) + unit" [style.line-height]="lineHeight + unit">
+      <ng-content></ng-content>
+    </div>
+  `,
+  styleUrls: ['text-overflow-ellipses.component.css']
+})
+
+export class TextOverflowEllipsesComponent {
+  @Input() numLines: number = 3;
+  @Input() lineHeight: number = 1.2;
+  @Input() unit: string = 'em';
```

Comment:
how come all the components have unit default to 'em' but then are set to 'px' whenever they're used?